apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs

```diff
@@ -103,14 +126,87 @@ public sealed class LoggerOptions
             MonitoredResource monitoredResource = null,
             BufferOptions bufferOptions = null,
             RetryOptions retryOptions = null,
-            TextWriter loggerDiagnosticsOutput = null)
+            TextWriter loggerDiagnosticsOutput = null) =>
+            CreateWithServiceContext(logLevel, logName, labels, monitoredResource, bufferOptions, retryOptions, loggerDiagnosticsOutput);
+
+        /// <summary>
+        /// Create a new instance of <see cref="LoggerOptions"/> with default service context.
+        /// If <paramref name="monitoredResource"/> is provided, the service context will be obtained from it.
+        /// Else, if running on GAE, the service context will be obtained from the platform.
+        /// </summary>
+        /// <param name="logLevel">Optional, the minimum log level.  Defaults to <see cref="LogLevel.Information"/></param>
+        /// <param name="logName">Optional, the name of the log.  Defaults to 'aspnetcore'.</param>
+        /// <param name="labels">Optional, custom labels to be added to log entries.
+        /// Keys and values added to <paramref name="labels"/> should not be null.
+        /// If they are, an exception will be throw when attempting to log an entry.
+        /// The entry won't be logged and the exception will be propagated depending
+        /// on the value of <see cref="RetryOptions.ExceptionHandling"/>.</param>
+        /// <param name="monitoredResource">Optional, the monitored resource.  The monitored resource will
+        /// be automatically detected if it is not set and will default to the global resource if the detection fails.
+        /// See: https://cloud.google.com/logging/docs/api/v2/resource-list </param>
+        /// <param name="bufferOptions">Optional, the buffer options.  Defaults to a <see cref="BufferType.Timed"/></param>
+        /// <param name="retryOptions">Optional, the retry options.  Defaults to a <see cref="RetryType.None"/></param>
+        /// <param name="loggerDiagnosticsOutput">Optional. If set some logger diagnostics info will be written
+        /// to the given <see cref="TextWriter"/>. Currently the only diagnostics info we provide is the URL where
+        /// the logs written with these options can be found.</param>
+        public static LoggerOptions CreateWithDefaultServiceContext(
```

Comment:
How about CreateWithDetectedServiceContext to make it clearer that this isn't the default behaviour?